pandas/core/indexes/multi.py

```diff
@@ -3045,9 +3045,11 @@ def _update_indexer(idxr, indexer=indexer):
                 indexer = Index(np.arange(n))
             if idxr is None:
                 return indexer
-            return indexer & idxr
+            return idxr & indexer
```

Comment:
I think we should remove _update_indexer entirely and simply set ```indexer = Index(np.arange(n))``` at the start of the loop

Reply:
I agree with the removal, it indeed make the code more clear.